### PR TITLE
Bugfix on sliders value 

### PR DESCRIPTION
**Pull request by nadijagraca**
## Description
Fixed bug on sliders and range sliders value 
Issue: https://github.com/mckinsey/vizro/issues/230

## Notice

- [x] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

  - I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
  - I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorized to submit this contribution on behalf of the original creator(s) or their licensees.
  - I certify that the use of this contribution as authorized by the Apache 2.0 license does not violate the intellectual property rights of anyone else.
  - I have not referenced individuals, products or companies in any commits, directly or indirectly.
  - I have not added data or restricted code in any commits, directly or indirectly.

### CODE REVIEW

File: vizro-core/changelog.d/20240115_110324_nadija_ratkusic_graca_slider_value_bug.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+<!--
+### Deprecated
+
+- A bullet item for the Deprecated category with a link to the relevant PR at the end of your entry, e.g. Enable feature XXX ([#1](https://github.com/mckinsey/vizro/pull/1))
+
+-->
+
+### Fixed
+
+- Fixed an issue with the default value of the `value` argument in sliders. ([#266](https://github.com/mckinsey/vizro/pull/266))
```
Comment (huong-li-nguyen) [inline vizro-core/changelog.d/20240115_110324_nadija_ratkusic_graca_slider_value_bug.md:40]:
```suggestion
- Fix bug of `value` argument not properly being evaluated in `Slider` and `RangeSlider`. ([#266](https://github.com/mckinsey/vizro/pull/266))
```

**Review by huong-li-nguyen — APPROVED** [review]
The example raised in the issue [here](https://github.com/mckinsey/vizro/issues/230) works for me now, but please wait for @maxschulz-COL approval to confirm whether this was the only issue 👍

File: vizro-core/src/vizro/models/_components/form/range_slider.py [hunk 2]
```diff
@@ -136,7 +136,11 @@ def build(self):
                                     if self.step
                                     else "slider_input_field_no_space_right",
                                 ),
-                                dcc.Store(id=f"temp-store-range_slider-{self.id}", storage_type="session"),
+                                dcc.Store(
+                                    id=f"{self.id}_input_store",
+                                    storage_type="session",
+                                    data=self.value or [value[0], value[1]],
+                                ),
```
Comment (petar-qb) [inline vizro-core/src/vizro/models/_components/form/range_slider.py:143]: At the beginning of the `build` method, a variable called `value` is calculated. So you can probably just reuse it and set it like this:
```
                                dcc.Store(
                                    id=f"{self.id}_input_store",
                                    storage_type="session",
                                    data=value,
                                ),
```

File: vizro-core/src/vizro/models/_components/form/slider.py [hunk 3]
```diff
@@ -112,7 +112,7 @@ def build(self):
                             persistence_type="session",
                             className="slider_input_field_right" if self.step else "slider_input_field_no_space_right",
                         ),
-                        dcc.Store(id=f"{self.id}_temp_store", storage_type="session"),
+                        dcc.Store(id=f"{self.id}_input_store", storage_type="session", data=self.value or self.min),
```
Comment (petar-qb) [inline vizro-core/src/vizro/models/_components/form/slider.py:115]: Based on the `RangeSlider.build` implementation, perhaps it makes sense to calculate variable `init_value = self.value or self.min` at the beginning of the `Slider.build` method and then just reuse it in three places below.

File: vizro-core/src/vizro/models/_components/form/range_slider.py (hunk 2 above)
Comment (petar-qb) [reply vizro-core/src/vizro/models/_components/form/range_slider.py:143]: We can also probably avoid ambiguity by renaming the `value` variable to something like `init_value` or whatever makes more sense to you.

**Review by petar-qb — COMMENTED** [review]
Thanks for the PR and renaming dcc.Store IDs 😄

**Review by petar-qb — APPROVED** [review]
It looks perfect now ⚡. 
However, I would wait for @maxschulz-COL approval.

File: vizro-core/src/vizro/models/_components/form/range_slider.py [hunk 4]
```diff
@@ -40,7 +40,7 @@ class RangeSlider(VizroBaseModel):
     min: Optional[float] = Field(None, description="Start value for slider.")
     max: Optional[float] = Field(None, description="End value for slider.")
     step: Optional[float] = Field(None, description="Step-size for marks on slider.")
-    marks: Optional[Dict[float, str]] = Field({}, description="Marks to be displayed on slider.")
+    marks: Optional[Dict[int, Union[str, Dict[str, str]]]] = Field({}, description="Marks to be displayed on slider.")
```
Comment (maxschulz-COL) [inline vizro-core/src/vizro/models/_components/form/range_slider.py:43]: I am not sure I understand this change. Did I miss something?

**Review by maxschulz-COL — APPROVED** [review]

File: vizro-core/src/vizro/models/_components/form/range_slider.py (hunk 4 above)
Comment (petar-qb) [inline vizro-core/src/vizro/models/_components/form/range_slider.py:43]: We should probably go with the following type:
`marks: Optional[Dict[int, Union[str, Dict[str, Any]]]]`

Comment (nadijagraca) [reply vizro-core/src/vizro/models/_components/form/range_slider.py:43]: This change is not connected to this bug fix, but I included it in this PR as it is fix on the sliders. This 'bug' was discovered yesterday by Peter when he tried to show marks on a RangeSlider. 
Basically, our previous type for `marks` which was `Optional[Dict[float, str]]` was disabling marks to be shown on the sliders as dcc sliders dictionary keys must be integers for the labels to be displayed. 

Additionally, dcc sliders have a handy way to styling individual marks, which was not supported by us so far, and that is achieved by providing marks values as a dictionary, something like below. :

To display marks on sliders I had to change float -> str, and to have this ability to style individual marks, i changed str -> Union[str, Dict[str, Any]]. 
`marks={1: {'label': "Jan 23", 'style': {'fontSize': '8px', "transform": "rotate(40deg)"}}}`

Screenshot 2024-01-16 at 18 10 54

**Pull request merged**